components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -279,9 +279,16 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
                     // Set OIDC Config Properties.
                     app.setRequestObjectSignatureValidationEnabled(
                             application.isRequestObjectSignatureValidationEnabled());
-                    app.setIdTokenEncryptionEnabled(application.isIdTokenEncryptionEnabled());
-                    app.setIdTokenEncryptionAlgorithm(application.getIdTokenEncryptionAlgorithm());
-                    app.setIdTokenEncryptionMethod(application.getIdTokenEncryptionMethod());
+
+                    // Validate IdToken Encryption configurations.
+                    if (application.isIdTokenEncryptionEnabled()) {
+                        app.setIdTokenEncryptionEnabled(application.isIdTokenEncryptionEnabled());
+                        if (application.isIdTokenEncryptionEnabled()) {
```

Comment:
Isn't this redundant?

Reply:
Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/commit/4adb2395c29dd91bf7b6d835c2bb878f96f3741e#diff-1e769d65ab3dba0fc2f6ab9b801f1e72